### ISSUE TITLE
Price and Group Price display incorrectly in product view and are not applied customer group tax

### ISSUE THREAD

**Issue by dattruongthe** [magento/magento2#26014]
### Preconditions (*)

1. Magento 2.3.3 
2. Installed via Composer

### Steps to reproduce (*)

1. Setup 2 Tax rates: **19%** and **0%**
2. Create 2 customer groups: **General** and **Retailer**
3. Create 1 product tax class: **Product 19%**
4. Setup 2 Tax rules: 
4.1 Rule 1: **Product 19%** - Group: **General** - Rate: **19%**
4.2 Rule 2: **Product 19%** - Group: **Retailer** - Rate: **0%**
<img width="1311" alt="Screen Shot 2019-12-12 at 21 14 41" src="https://user-images.githubusercontent.com/17728749/70719362-72300800-1d24-11ea-8e59-db27a1e5a53b.png">

5. Create Simple product with price: **100$** and Tax Class: **Product 19%**

6. Create a customer and assign to **Retailer** group

7. Store -> Configuration -> Sales -> Tax is setup as follow:
<img width="794" alt="Screen Shot 2019-12-12 at 21 11 41" src="https://user-images.githubusercontent.com/17728749/70719170-11a0cb00-1d24-11ea-9dcc-840264e68b86.png">
<img width="748" alt="Screen Shot 2019-12-12 at 21 11 55" src="https://user-images.githubusercontent.com/17728749/70719178-14032500-1d24-11ea-863f-cc4d7b4116c2.png">

### Expected result (*)

After logged in to **Retailer Account**, the price should be displayed:
1. Product view: 84.03$

### Actual result (*)

After logged in to **Retailer Account**, the price is displayed
1. Product view: 100$
<img width="541" alt="Screen Shot 2019-12-12 at 21 10 28" src="https://user-images.githubusercontent.com/17728749/70718995-d7cfc480-1d23-11ea-8d69-33add378ce07.png">
2. After added product to cart: the price is shown 84.03$ which is correct
<img width="1249" alt="Screen Shot 2019-12-12 at 21 10 54" src="https://user-images.githubusercontent.com/17728749/70719024-e61de080-1d23-11ea-969a-c77de07aa809.png">

### Steps to reproduce (*)
1. Add Group Price for **Retailer Group** is: 90$ (use same product as above).
### Expected result (*)
1. Product view shows 2 prices: Regular price: 84.03$ - Special price 75.63$
### Actual result (*)
1. Product view shows 2 prices: Regular price: 100$ - Special price 90$
<img width="537" alt="Screen Shot 2019-12-12 at 21 13 40" src="https://user-images.githubusercontent.com/17728749/70719310-59275700-1d24-11ea-908f-f862ee05b4e7.png">
2. After added product to cart: the price is shown 75.63$ which is correct
<img width="1259" alt="Screen Shot 2019-12-12 at 21 13 51" src="https://user-images.githubusercontent.com/17728749/70719329-604e6500-1d24-11ea-8f9a-bbd3fdcca628.png">

**Comment by dattruongthe** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @dattruongthe. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @dattruongthe, here is your [Magento instance](https://i-26014-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26014-2-4-develop.instances.magento-community.engineering/admin](https://i-26014-2-4-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by dattruongthe** [comment]
Confirm issue is fixed on magento 2.4 instance

**Comment by dattruongthe** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @dattruongthe. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by dattruongthe** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @dattruongthe. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by dattruongthe** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @dattruongthe. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by krishprakash** [comment]
Hi @dattruongthe. Thank you for your report.
Not able to reproduce the issue on latest 2.4 develop instance
So closing this issue
Thanks.

**Comment by dattruongthe** [comment]
Hi @krishprakash , Thanks for your checking.
This issue just happens on Magento version <= 2.3.3 and was fixed for Magento 2.4. For those who are getting the same issue, you can fix by updating this file: `Magento/Tax/Model/Config.php`

```
public function needPriceConversion($store = null)
    {
        $res = false;
        $priceIncludesTax = $this->priceIncludesTax($store) || $this->getNeedUseShippingExcludeTax();
        if ($priceIncludesTax) {
            switch ($this->getPriceDisplayType($store)) {
                case self::DISPLAY_TYPE_EXCLUDING_TAX:
                case self::DISPLAY_TYPE_BOTH:
                    return self::PRICE_CONVERSION_MINUS;
                case self::DISPLAY_TYPE_INCLUDING_TAX:
                    $res = $this->displayCartPricesInclTax($store);
                    break;
                default:
                    break;
            }
        } else {
            switch ($this->getPriceDisplayType($store)) {
                case self::DISPLAY_TYPE_INCLUDING_TAX:
                case self::DISPLAY_TYPE_BOTH:
                    return self::PRICE_CONVERSION_PLUS;
                case self::DISPLAY_TYPE_EXCLUDING_TAX:
                    $res = false;
                    break;
                default:
                    break;
            }
        }

        if ($res === false) {
            $res = $this->displayCartPricesBoth();
        }
        return $res;
    }
```
